### PR TITLE
:sparkles: update logic in side event arranger.

### PR DESCRIPTION
**Pull request by PRBaraiya**
Fixes issue #130 .

### CODE REVIEW

File: lib/src/event_arrangers/side_event_arranger.dart [hunk 1]
```diff
@@ -16,145 +16,88 @@ class SideEventArranger<T extends Object?> extends EventArranger<T> {
...
-            arrangedEvent[index] = arrangedEvent[index]
-                .getWithUpdatedRight(arrangedEvent[index].right - widthPerCol);
-          }
-        } else {
+      final slotWidth = width / column;
+
+      for (final sideEvent in sideEventData) {
+        if (sideEvent.event.startTime == null ||
+            sideEvent.event.endTime == null) {
+          debugPrint("Start time or end time of an event can not be null. "
```
Comment (HappyMakadiyaS) [inline lib/src/event_arrangers/side_event_arranger.dart:71]: use assert

File: lib/src/event_arrangers/side_event_arranger.dart [hunk 2]
```diff
@@ -16,145 +16,88 @@ class SideEventArranger<T extends Object?> extends EventArranger<T> {
...
 
-      _insertIntoTable(table, durations, rowCounter, tempEvents[0]);
+      var column = 1;
+      final sideEventData = <_SideEventData<T>>[];
+      var currentEventIndex = 0;
 
-      tempEvents.removeAt(0);
+      while (concurrentEvents.isNotEmpty) {
+        final event = concurrentEvents[currentEventIndex];
+        final end = event.endTime!.getTotalMinutes;
```
Comment (HappyMakadiyaS) [inline lib/src/event_arrangers/side_event_arranger.dart:46]: handle null check

File: lib/src/event_arrangers/side_event_arranger.dart [hunk 3]
```diff
@@ -16,145 +16,88 @@ class SideEventArranger<T extends Object?> extends EventArranger<T> {
...
+        concurrentEvents.removeAt(currentEventIndex);
 
-      while (tempEvents.isNotEmpty && eventCounter < tempEvents.length) {
-        if ((tempEvents[eventCounter].startTime?.getTotalMinutes ?? 0) > end) {
-          _insertIntoTable(
-              table, durations, rowCounter, tempEvents[eventCounter]);
+        while (currentEventIndex < concurrentEvents.length) {
+          if (end <
+              concurrentEvents[currentEventIndex].startTime!.getTotalMinutes) {
+            break;
```
Comment (HappyMakadiyaS) [inline lib/src/event_arrangers/side_event_arranger.dart:53]: ^

**Review by HappyMakadiyaS — CHANGES_REQUESTED** [review]

File: lib/src/event_arrangers/side_event_arranger.dart (hunk 2 above)
Comment (PRBaraiya) [reply lib/src/event_arrangers/side_event_arranger.dart:46]: There will be no events with null endTime. As we are already removing events with null endTime or startTime in MergeEvent Arranger at line no 19.

File: lib/src/event_arrangers/side_event_arranger.dart (hunk 3 above)
Comment (PRBaraiya) [reply lib/src/event_arrangers/side_event_arranger.dart:53]: There will be no events with null endTime. As we are already removing events with null endTime or startTime in MergeEvent Arranger at line no 19.

**Review by HappyMakadiyaS — APPROVED** [review]

**Pull request merged**